### ISSUE TITLE
[Bug]: Bottles freezes after every launch after the last update

### ISSUE THREAD

**Issue by Protrikk** [bottlesdevs/Bottles#2354]
### Describe the bug

![obraz](https://user-images.githubusercontent.com/42098407/206521044-7cde0f78-2082-45e1-afc7-fc1ac3910a1e.png)
Bottles is either stuck on "Downloading 1/3 components" or show the above icon and immidietly freezes after the last update.

### To Reproduce

1. Open bottles
2. Wait for it to launch
3. Freezes, can't even move the windows around

### Package

Flatpak from Flathub

### Distribution

OpenSUSE microos

### Troubleshooting Logs

```shell
Version: 2022.11.14
Display:
    X.org: true
    X.org (port): :0
    Wayland: true
Graphics:
    vendors:
        nvidia:
            vendor: nvidia
            envs:
                __NV_PRIME_RENDER_OFFLOAD: '1'
                __GLX_VENDOR_LIBRARY_NAME: nvidia
                __VK_LAYER_NV_optimus: NVIDIA_only
            icd: /usr/lib/x86_64-linux-gnu/GL/vulkan/icd.d/nvidia_icd.json:/usr/lib/i386-linux-gnu/GL/vulkan/icd.d/nvidia_icd.json
    prime:
        integrated: null
        discrete: null
Kernel:
    Type: Linux
    Version: 6.0.10-1-default
Distro:
    Name: GNOME
    Version: '43'
Disk:
    Total: 16772526080
    Free: 16772370432
RAM:
    MemTotal: "33,5\u202FGB"
    MemAvailable: "28,0\u202FGB"
Bottles_envs: null

```

### Display Protocol

Wayland

### Display Backend used by Bottles

Wayland

### Additional context

I'm currently using a hotspot from my phone which gnome software detects as a "measured connection", not sure if this has anything to do with it though

**Comment by Protrikk** [comment]
News: after restarting it enough times it seems to have downloaded "1 out of 3 components" and now is stuck on "launching..."

**Comment by Protrikk** [comment]
Restarting it after that it seems to have launched fine, sorry for a bit of a random issue but I just wanted to play a video game and got annoyed. I think stuff like this should still be resolved though so I'm not closing it.

**Comment by mirkobrombin** [comment]
Can't replicate

**Comment by Protrikk** [comment]
I'm not sure if I'm able to but from what I understand, bottles went bananas when a components update was available. I'm not able to replicate it now either since after those restarts it finally managed to update and launch

**Comment by mirkobrombin** [comment]
I'll keep this open for a few days

**Comment by Protrikk** [comment]
I'll try to clear things up a bit, hopefully some of this does:
- Bottles either started frozen (just show the no internet icon and froze completely) or semi-frozen, I'm assuming trying to download the updates
- If it started in the kinda frozen state, It'd download them without updating the "1 out of 3" text and try to start fully afterwards, but instead stayed on the download screen with just the "launching..." text
- Starting it again after it finished the downloads made it work again

I have relitevly slow internet (at that particular moment 1.5mbps peak download speed, it's all over the place), maybe that's something to do with that?

**Comment by StoneMoe** [comment]
> I have relitevly slow internet (at that particular moment 1.5mbps peak download speed, it's all over the place), maybe that's something to do with that?

Bottles is heavily relies on good internet connection from my experience, and UI design is not always good at syncing with its working state.
Try block Bottles' network before open it if you really have a bad network.

**Comment by Cyber-Oto** [comment]
Since yesterday, I have similar issue (same version as OP) - when I open the app, it is stuck on "Starting up..." window with "Fetched 3 of 3 packages / Downloading ~20kb of packages..." message.

I left it for a few hours running and nada, no joy. 

It was working OK earlier that day. Now I get this each time I try to start it.

**Comment by Kinsteen** [comment]
@Cyber-Oto Can you launch Bottles from the command line and send the logs here please?

**Comment by Cyber-Oto** [comment]
System:

```
  Kernel: 5.15.81-1-MANJARO arch: x86_64 bits: 64 compiler: gcc v: 12.2.0
    Desktop: GNOME v: 43.1 tk: GTK v: 3.24.35 wm: gnome-shell dm: GDM v: 43.0
    Distro: Manjaro Linux base: Arch Linux
```

Please find the console output:

```
09:08:47 (INFO) Bottles Started! 
09:08:48 (INFO) Catalog components loaded 
09:08:48 (INFO) Catalog installers loaded 
09:08:48 (INFO) Catalog dependencies loaded 
09:08:48 (INFO) Performing Bottles checks… 
09:08:48 (INFO) Dxvks found:
	 - dxvk-1.10.3
	 - dxvk-2.0
 
09:08:48 (INFO) Vkd3ds found:
	 - vkd3d-proton-2.8
	 - vkd3d-proton-2.6-1-5b73139
 
09:08:48 (INFO) Nvapis found:
	 - dxvk-nvapi-v0.6
	 - dxvk-nvapi-v0.5.4
 
09:08:48 (INFO) Latencyflexs found:
	 - latencyflex-v0.1.0
 
09:08:48 (INFO) Runners found:
	 - soda-7.0-7
	 - soda-7.0-8
 
09:08:48 (INFO) Temp directory cleaned successfully! 
09:08:48 (INFO) Bottles found:
	 - Epic
	 - EA
	 - Origin
	 - Blizzard
 
(bottles:85199): Gtk-CRITICAL **: 09:08:48.759: gtk_widget_set_parent: assertion '_gtk_widget_get_parent (widget) == NULL' failed
09:08:48 (INFO) Library saved 
Traceback (most recent call last):
  File "/usr/share/bottles/bottles/frontend/windows/main_window.py", line 155, in set_manager
    self.page_library = LibraryView(self)
  File "/usr/share/bottles/bottles/frontend/views/library.py", line 43, in __init__
    self.update()
  File "/usr/share/bottles/bottles/frontend/views/library.py", line 56, in update
    entry = LibraryEntry(self, u, e)
  File "/usr/share/bottles/bottles/frontend/widgets/library.py", line 62, in __init__
    self.__remove_entry()
  File "/usr/share/bottles/bottles/frontend/widgets/library.py", line 137, in __remove_entry
    self.library.remove_entry(self)
  File "/usr/share/bottles/bottles/frontend/views/library.py", line 66, in remove_entry
    entry.hide()
  File "/usr/share/bottles/bottles/frontend/widgets/library.py", line 169, in hide
    self.get_parent().set_visible(False)
AttributeError: 'NoneType' object has no attribute 'set_visible'
```

**Comment by Kinsteen** [comment]
Thank you. Can you send your library.yml file located in your Bottles folder? It seems to be a bug with the library loading. After you saved it, you can try to delete it and launch Bottles again

**Comment by Cyber-Oto** [comment]
Removing the `library.yml` file helped and now the app opens, as expected

Thank you!

```
(bottles:86471): Gtk-CRITICAL **: 09:21:02.155: gtk_widget_set_parent: assertion '_gtk_widget_get_parent (widget) == NULL' failed
09:21:02 (WARNING) Library file not found, creating new one 
09:21:02 (INFO) Library saved 
10:21:02 (INFO) Library saved 
```

Note: it seems that now one of my bottles is not loaded or listed in the console, but I will deal with that.

Could it be that this bottle is "broken" and causing this startup issue?

**Comment by mirkobrombin** [comment]
I tried to replicate this in many ways without success. Is there a possibility that you skipped an update and I was careless not to handle a library format change migration?

**Comment by Cyber-Oto** [comment]
In my case, it seems I was the root cause - I was trying to install vcredist2012 dependency to that bottle. It was taking a long time without any visible action, and I recall closing the app.

A bit later, I have noticed that bottle `.fvs` folder was quite big (10+ Gb), compared to other bottles. I have re-created the bottle (_minus the vcredist part_) and it's working fine now.

**Comment by torbennehmer** [comment]
I've seen this as well, it was too a matter of a broken library.yml.
I have been toying around with a bottle, that was added to the library. Did remove/add stuff for a bit. However, I unfortunately can't really say, what I did before things broke. It might even be that I deleted the bottle, I also had some weird freezes of bottles and its wine instance (was to bold when trying out cutting-edge stuff).
